recipes/pdal/all/conanfile.py

```diff
@@ -0,0 +1,100 @@
+import os
+from conans import ConanFile, tools, CMake
+from conans.errors import ConanInvalidConfiguration
+
+class PdalConan(ConanFile):
+    name = "pdal"
+    description = "PDAL is Point Data Abstraction Library. GDAL for point cloud data."
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://pdal.io"
+    license = "BSD"
+    exports_sources = ["CMakeLists.txt", "patches/*"]
+    generators = "cmake", "cmake_find_package"
+    settings = "os", "arch", "compiler", "build_type"
+    options = {"shared": [True, False],
+               "fPIC": [True, False],
+               "with_xml": [True, False],
+               "with_zstd": [True, False],
+               "with_laszip": [True, False]}
+    default_options = {"shared": False,
+                       "fPIC": True,
+                       "with_xml": True,
+                       "with_zstd": True,
+                       "with_laszip": True}
+    topics = ("conan", "pdal", "gdal")
+
+    _source_subfolder = "source_subfolder"
+
+    def requirements(self):
+        self.requires("gdal/3.1.4")
+        self.requires("libgeotiff/1.6.0")
+        if self.options.with_xml:
+            self.requires("libxml2/2.9.10")
+        if self.options.with_zstd:
+            self.requires("zstd/1.4.5")
+        if self.options.with_laszip:
+            self.requires("laszip/3.4.3")
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        # upstream export/install targets do not work with static builds
+        if not self.options.shared:
+            raise ConanInvalidConfiguration("pdal does not support building as a static lib yet")
+        if self.options.shared:
+            del self.options.fPIC
+        if self.settings.compiler.cppstd:
+            tools.check_min_cppstd(self, 11)
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        os.rename("PDAL-%s-src" % self.version, self._source_subfolder)
+
+    def _configure_cmake(self):
+        cmake = CMake(self)
```

Comment:
This project seems large enough, did you consider caching the cmake object?

Reply:
Done